### DISCUSSION TITLE
Continuous Update of Canvas Lat/Long

### DISCUSSION THREAD

**Discussion by linhtv97** [RodrigoHamuy/react-three-map#132] [General]
Hi all,  

In my project, I’m syncing the `Canvas` position with the map center whenever the center updates. However, with complex scenes (many objects), the updates cause the view to "jump" or behave inconsistently.  

Has anyone faced this issue? Are there solutions to smooth or optimize the updates to avoid the jumping effect?  

Looking forward to your suggestions!  

`
Canvas
            overlay={false}
            latitude={viewState.latitude}
            longitude={viewState.longitude}
            shadows
            onCreated={({ gl }) => {
              gl.setClearColor(0x000000)
              gl.setClearAlpha(0)
              gl.autoClear = false
              gl.shadowMap.enabled = true
              gl.shadowMap.needsUpdate = true
              gl.shadowMap.type = THREE.PCFSoftShadowMap
            }}
            gl={{ alpha: false, antialias: true }}
            camera={{ position: [0, 500, 0], far: 5000 }}
            frameloop='demand'
          >`
          
          
And I use `NearCoordinates` to display Object3D
          
          
          

https://github.com/user-attachments/assets/e1c5af3a-2624-40d9-a75f-ae033beaf0ce

**Comment by RodrigoHamuy** [comment]
When you say _"when the center updates"_ you mean `latitude={viewState.latitude} longitude={viewState.longitude}` are changing values?
If so, this is an easy fix!
Just avoid doing that, so that the coordinates don't need to be transformed to threejs coordinates every time you move the camera.

**Reply by linhtv97** [reply]
> When you say "when the center updates" you mean latitude={viewState.latitude} longitude={viewState.longitude} are changing values?

Yes,
I have to update the position of the sun every time the camera moves, so I need to update the three.js coordinates as well
